### ISSUE TITLE
Component parameters not generated automatically?

### ISSUE THREAD

**Issue by mfeurer** [openml/OpenML#280]
I am uploading a flow with a component to the test server. The flow is generated correctly with the component and the parameters, but the parameters of the component are not added to the component element of the server. These are the IDs of the generated flows: 3186, 3187 and you can find the xml below:

```
<oml:flow xmlns:oml="http://openml.org/openml">
	    <oml:name>sklearn.grid_search.RandomizedSearchCV(sklearn.linear_model.logistic.LogisticRegression)</oml:name>
	    <oml:external_version>sklearn_0.17.1</oml:external_version>
	    <oml:description>Automatically generated.</oml:description>
	    <oml:parameter>
		    <oml:name>cv</oml:name>
	    </oml:parameter>
	    <oml:parameter>
		    <oml:name>error_score</oml:name>
		    <oml:default_value>raise</oml:default_value>
	    </oml:parameter>
	    <oml:parameter>
		    <oml:name>estimator</oml:name>
		    <oml:default_value>sklearn.linear_model.logistic.LogisticRegression</oml:default_value>
	    </oml:parameter>
	    <oml:parameter>
		    <oml:name>fit_params</oml:name>
	    </oml:parameter>
	    <oml:parameter>
		    <oml:name>iid</oml:name>
		    <oml:default_value>True</oml:default_value>
	    </oml:parameter>
	    <oml:parameter>
		    <oml:name>n_iter</oml:name>
		    <oml:default_value>5</oml:default_value>
	    </oml:parameter>
	    <oml:parameter>
		    <oml:name>n_jobs</oml:name>
		    <oml:default_value>1</oml:default_value>
	    </oml:parameter>
	    <oml:parameter>
		    <oml:name>param_distributions</oml:name>
		    <oml:default_value>{'C': [1e-05, 0.0001, 0.001, 0.01, 0.1, 1, 10, 100, 1000, 10000]}</oml:default_value>
	    </oml:parameter>
	    <oml:parameter>
		    <oml:name>pre_dispatch</oml:name>
		    <oml:default_value>2*n_jobs</oml:default_value>
	    </oml:parameter>
	    <oml:parameter>
		    <oml:name>random_state</oml:name>
	    </oml:parameter>
	    <oml:parameter>
		    <oml:name>refit</oml:name>
		    <oml:default_value>True</oml:default_value>
	    </oml:parameter>
	    <oml:parameter>
		    <oml:name>scoring</oml:name>
	    </oml:parameter>
	    <oml:parameter>
		    <oml:name>verbose</oml:name>
	    </oml:parameter>
	    <oml:component>
		    <oml:identifier>estimator</oml:identifier>
		    <oml:flow xmlns:oml="http://openml.org/openml">
			    <oml:name>sklearn.linear_model.logistic.LogisticRegression</oml:name>
			    <oml:external_version>sklearn_0.17.1</oml:external_version>
			    <oml:description>Automatically created sub-component.</oml:description>
			    <oml:parameter>
				    <oml:name>C</oml:name>
				    <oml:default_value>1.0</oml:default_value>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>class_weight</oml:name>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>dual</oml:name>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>fit_intercept</oml:name>
				    <oml:default_value>True</oml:default_value>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>intercept_scaling</oml:name>
				    <oml:default_value>1</oml:default_value>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>max_iter</oml:name>
				    <oml:default_value>100</oml:default_value>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>multi_class</oml:name>
				    <oml:default_value>ovr</oml:default_value>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>n_jobs</oml:name>
				    <oml:default_value>1</oml:default_value>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>penalty</oml:name>
				    <oml:default_value>l2</oml:default_value>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>random_state</oml:name>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>solver</oml:name>
				    <oml:default_value>liblinear</oml:default_value>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>tol</oml:name>
				    <oml:default_value>0.0001</oml:default_value>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>verbose</oml:name>
			    </oml:parameter>
			    <oml:parameter>
				    <oml:name>warm_start</oml:name>
			    </oml:parameter>
		    </oml:flow>
	    </oml:component>
    </oml:flow>
```

**Comment by janvanrijn** [comment]
Can you please double check the flow id? The database does not contain these flows. Or attach the flow xml with the issue, as it is hard to parse this one

**Comment by mfeurer** [comment]
I have updated the formatting of the xml. It's true, I might have deleted the flow. After working with flows the last few days, I think the only problem is that the frontend and the json download don't show parameters of automatically created flows. See for example this flow in the frontend http://capa.win.tue.nl/f/3867 and compare it to the xml description.

**Comment by joaquinvanschoren** [comment]
Can you please check again? I believe this has been fixed in the mean time...

**Comment by mfeurer** [comment]
Apparently, they weren't displayed in the front end. Fixed now.
